index.rst

```diff
@@ -1,29 +1,492 @@
+The Notebook Aspect (and only it) will therefore get a notebook token rather than an internal token.
+This is a special case of an internal token that has all of the same scopes as the user's original session token, and is associated with the user's notebook server.
+It may have a lifetime limited to the lifetim of the user's notebook server.
+
+``Authorization`` headers used for token authentication should be (but are not yet) filtered out of the request so that they are not passed down to the underlying Science Platform service.
+Otherwise, a service could recover the user's original token from the HTTP headers of the request.
+
+.. _service-auth:
+
+Service-to-service authentication
+---------------------------------
+
+In some cases, services may need to access other Science Platform services on their own behalf, unrelated to a user request.
+For example, a monitoring system may need to make periodic requests to authenticated APIs of Science Platform services to ensure that they are running and correctly responding to requests.
+
+These requests will be authorized in the same way as subrequests discussed above, by interposing the same authentication system used for user requests.
+They are authenticated with service tokens, which are issued only to services and are never used by users.
+Services can ask for service tokens by creating a custom Kubernetes resource specifying the properties of the service token, including the identity of the service and the scopes it requires.
+The authentication service will then provide that service token as a Kubernetes ``Secret`` resource associated with the request in the custom resource, and thereby make it available to the service pods through the normal Kubernetes mechanisms for injecting secrets into pods.
+The authentication service will also automatically refresh the service token to ensure that it does not expire.
+
+As specified in DMTN-225_, the usernames associated with all such tokens must begin with ``bot-``.
+
+OpenID Connect authentication
+-----------------------------
+
+Some Science Platform deployments run third-party services (Chronograf_, for example) that themselves want to do OpenID Connect authentication of the user.
+To support those services, the authentication service of the Science Platform is also an OpenID Connect provider.
+Such services can then point to the authentication service as the authentication provider, and those authentications will use the same source of identity as other authentications to the Science Platform.
+(This authentication is independent of any use of OpenID Connect by the authentication service to a federated or local identity provider external to the Science Platform, although the two authentications will be chained together when needed.)
+
+.. _Chronograf: https://www.influxdata.com/time-series-platform/chronograf/
+
+At present, OpenID Connect authentication used in this fashion does not do any access control.
```

Comment:
Is there an actual path to doing otherwise?

Reply:
Yes: we could associate OpenID Connect clients with required scopes and refuse to authenticate the user if they're missing that scope. There will be more discussion of this in SQR-069.